charts/router/templates/router-deployment.yaml

```diff
@@ -1,3 +1,4 @@
+{{- if not .Values.global.experimental_native_ingress }}
```

Comment:
Ah. I am sorry for a bit of mis-understanding here. I'd forgotten that `global` was a special scope within Helm that allowed charts to "look up." I've now realized that...

I understand what you're trying to do here. Regardless of whether someone installs the router chart directly _or_ installs the Workflow chart, which includes the router chart, a value for `global.experimental_native_ingress` should be available.

Although I understand what you're trying to do and agree it would work, I think there are a couple smells here... and I think there's a much cleaner alternative.

The smells:

The router chart now has the ability, when installed, depending on the value of `global.experimental_native_ingress` to actually _not_ install any of its k8s resources. i.e. It is possible to effect a "no-op" install of the router. _(Install this chart-- but dont!)_ This doesn't make sense _unless_ considered in the broader context of re-use by the Workflow chart. That means the router chart has been designed with that specific re-use by the Workflow chart in mind. And that's a smell. Because, what really, does "experimental native ingress" mean to the router itself? Nothing, in fact.

As I've said before, I think the router requires no changes to support this experiment. (Neither Go nor chart changes are in order.) All that needs to be done is for the Workflow chart to _conditionally_ include or not include the router chart. Until re-reading the Helm documentation this evening, I'd hoped, but hadn't known for sure that was possible... it is...

This section of the documentation shows two techniques that could be used to conditionally include or exclude the router chart in its entirety from a Workflow installation:

https://github.com/kubernetes/helm/blob/master/docs/charts.md#tags-and-condition-fields-in-requirementsyaml